### PR TITLE
feat: add -v (--version) options to both Cli and Server

### LINKED ISSUE #397: Add version option to display UC version on command line
When executed with `--version`, `start-uc-server` and `uc` command should display the UC version:

```
❯ ./bin/start-uc-server --version

Parsing Failed. Reason: Unrecognized option: --version
```

```
❯ ./bin/uc --version
Error occurred while parsing the command. Please check the command and try again. Missing argument for option: version
```

### PR DESCRIPTION
**Pull request by SemyonSinchenko**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**
Close #397

Motivation:
As described in the issue.

Changes:

- Add `-v` / `--version` to Server Cli;
- Add `-v` / `--version` to Cli;
- Update `build.sbt` to add `VersionUtils` to Cli;
- Update `test_tutorial.py` to check that `-v` works;

### CODE REVIEW

File: examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java [hunk 1]
```diff
@@ -67,6 +68,12 @@ public static void main(String[] args) {
         return;
       }
 
+      // Check if version option is provided
```
Comment (jaceklaskowski) [inline examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java:71]: Remove the comment = the code says it all

File: examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java [hunk 2]
```diff
@@ -532,6 +532,10 @@ public static void printHelp() {
         "For detailed help on entity specific operations, use bin/uc <entity> --help");
   }
 
+  public static void printVersion() { // This class is generated during the build
```
Comment (jaceklaskowski) [inline examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java:535]: Remove the comment

File: examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java [hunk 3]
```diff
@@ -532,6 +532,10 @@ public static void printHelp() {
...
+    System.out.printf("UnityCatalog CLI Version: %s\n", VersionUtils.VERSION);
```
Comment (jaceklaskowski) [inline examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java:536]: Can you just print out the version alone?

```java
System.out.println(VersionUtils.VERSION);
```

File: server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java [hunk 4]
```diff
@@ -146,9 +146,15 @@ public static void main(String[] args) {
             .desc("Port number to run the server on. Default is 8080.")
             .type(Integer.class)
             .build());
+    options.addOption(
+        Option.builder("v").longOpt("version").hasArg(false).desc("Print version").build());
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java:150]: s/Print/Display the

File: server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java [hunk 5]
```diff
@@ -146,9 +146,15 @@ public static void main(String[] args) {
...
     CommandLineParser parser = new DefaultParser();
     try {
       CommandLine cmd = parser.parse(options, args);
+      if (cmd.hasOption("v")) {
+        System.out.printf("Unity Catalog Version: %s\n", VersionUtils.VERSION);
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java:155]: Please print out the version alone

**Review by jaceklaskowski — CHANGES_REQUESTED** [review]

**Comment by jaceklaskowski** [comment]
> P.S. It is impossible to add --version to Cli due to collision with io.unitycatalog.cli.utils.CliParams that already contains VERSION (in a different meaning).

That has to be fixed, too. I'm sure users will get annoyed by this "inconsistency". Whatever entity uses `--version` should change it to `--[entity]_version`.

**Comment by SemyonSinchenko** [comment]
> > P.S. It is impossible to add --version to Cli due to collision with io.unitycatalog.cli.utils.CliParams that already contains VERSION (in a different meaning).
> 
> That has to be fixed, too. I'm sure users will get annoyed by this "inconsistency". Whatever entity uses `--version` should change it to `--[entity]_version`.


@jaceklaskowski Thank you for the review!

About your comment. It is used by models. `version` is an attribute of the `ModelVersion` object. I can change the API from `--version` to `--model-version` (or should it be `--model_version`?). But it will require also to change [public Api](https://github.com/unitycatalog/unitycatalog/blob/main/api/Models/ModelVersionInfo.md), so it will be a breaking change. As well as updates in [`all.yaml`](https://github.com/unitycatalog/unitycatalog/blob/main/api/all.yaml#L1926). I can do it, of course. And it is an interesting task. I just want to be sure before doing so big and user-facing change.

**Comment by SemyonSinchenko** [comment]
@jaceklaskowski I implemented it in the following way:
- version option does not require args anymore;
- that allows to check options and in the case of no args version is printed;
- in the case of args are provided I explicitly check `option.hasArg` for `version` (if present) and print the same message that we before;
- in other cases behavior is identical to the main;

P.S. Looks like a terrible workaround. What do you think about planning a breaking change for the future minor release and refactor `version` -> `model_version`?

**Comment by jaceklaskowski** [comment]
> What do you think about planning a breaking change for the future minor release and refactor version -> model_version?

That's inevitable (given that we've got `model_name` so why not `model_version` in the API?!) Can you report an issue to bring it up?

File: examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java [hunk 6]
```diff
@@ -28,8 +28,15 @@ public static void main(String[] args) {
...
-                options.addOption(Option.builder().longOpt(cliParam.val()).hasArg().build()));
+                options.addOption(
+                    Option.builder()
+                        .longOpt(cliParam.val())
+                        .optionalArg(cliParam.val().equals("version"))
+                        .hasArg() // See
+                        // https://github.com/unitycatalog/unitycatalog/pull/398#issuecomment-2325039123
+                        .build()));
     options.addOption("h", "help", false, "Print help message.");
+    options.addOption("v", false, "Print the version of UC.");
```
Comment (jaceklaskowski) [inline examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java:39]: Rephrase:

> Display the version of Unity Catalog CLI

File: examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java [hunk 7]
```diff
@@ -67,6 +74,34 @@ public static void main(String[] args) {
...
+      // To allow that behaviour version does not require arg, but if it used for entity args should
+      // be checked!
+      for (Option option : cmd.getOptions()) {
+        if (option.getLongOpt().equals("version")) {
+          if (cmd.getArgs().length == 0) {
+            CliUtils.printVersion();
+            return;
+          } else {
+            if (option.getValue() == null) {
+              System.out.println(
```
Comment (jaceklaskowski) [inline examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java:94]: Should this go to `err` instead?

File: server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java [hunk 8]
```diff
@@ -152,9 +152,15 @@ public static void main(String[] args) {
             .desc("Port number to run the server on. Default is 8080.")
             .type(Integer.class)
             .build());
+    options.addOption(
+        Option.builder("v").longOpt("version").hasArg(false).desc("Display the version").build());
```
Comment (jaceklaskowski) [inline server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java:156]: Display the version of the Unity Catalog server

**Review by jaceklaskowski — CHANGES_REQUESTED** [review]
Looks so awesome! Thanks for the changes. There are some nits waiting to be addressed but it's super dope already 👍

File: examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java (hunk 7 above)
Comment (SemyonSinchenko) [reply examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java:94]: It mimics exactly the behavior of apache commons internal validator (the error, when the arg is required but not provided). I checked it: it goes to stdout, not stderr (I tried to redirect 1>out 2>err and in both cases of my "validation" and validation from apache commons all the output is in out)

**Comment by SemyonSinchenko** [comment]
@jaceklaskowski Hi! This PR is ready for review, may you please take a look when you have time? Thanks in advance!

**Review by jaceklaskowski — APPROVED** [review]
LGTM 👍

**Comment by jaceklaskowski** [comment]
@dennyglee Can you have a look at this PR? 🙏

**Review by dennyglee — APPROVED** [review]
LGTM - thanks a bunch!

**Pull request merged**